server/admin/api.go

```diff
@@ -107,6 +108,98 @@ func (s *Server) apiMarketInfo(w http.ResponseWriter, r *http.Request) {
 	writeJSON(w, mktStatus)
 }
 
+// apiMarketOrderBook is the handler for the '/market/{marketName}/orderbook'
+// API request.
+func (s *Server) apiMarketOrderBook(w http.ResponseWriter, r *http.Request) {
+	mkt := strings.ToLower(chi.URLParam(r, marketNameKey))
+	status := s.core.MarketStatus(mkt)
+	if status == nil {
+		http.Error(w, fmt.Sprintf("unknown market %q", mkt), http.StatusBadRequest)
+		return
+	}
+	if !status.Running {
+		http.Error(w, fmt.Sprintf("market %q not running", mkt), http.StatusBadRequest)
+		return
+	}
+	orders, err := s.core.BookOrders(status.Base, status.Quote)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to obtain order book: %v", err), http.StatusInternalServerError)
+		return
+	}
+	msgBook := make([]*msgjson.BookOrderNote, 0, len(orders))
+	for _, o := range orders {
+		msgBook = append(msgBook, market.LimitOrderToMsgOrder(o, mkt))
+	}
+	msg := &msgjson.OrderBook{
+		MarketID: mkt,
+		Epoch:    uint64(status.ActiveEpoch),
+		Orders:   msgBook,
+	}
+	writeJSON(w, msg)
+}
+
+// hander for route '/market/{marketName}/epochorders' API request.
+func (s *Server) apiMarketEpochOrders(w http.ResponseWriter, r *http.Request) {
+	mkt := strings.ToLower(chi.URLParam(r, marketNameKey))
+	status := s.core.MarketStatus(mkt)
+	if status == nil {
+		http.Error(w, fmt.Sprintf("unknown market %q", mkt), http.StatusBadRequest)
+		return
+	}
+	if !status.Running {
+		http.Error(w, fmt.Sprintf("market %q not running", mkt), http.StatusBadRequest)
+		return
+	}
+	los, mos, err := s.core.EpochOrders(status.Base, status.Quote)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to obtain epoch orders: %v", err), http.StatusInternalServerError)
+		return
+	}
+	msgBook := make([]*msgjson.BookOrderNote, 0, len(los)+len(mos))
+	for _, o := range los {
+		msgBook = append(msgBook, market.LimitOrderToMsgOrder(o, mkt))
+	}
+	for _, o := range mos {
+		msgBook = append(msgBook, market.MarketOrderToMsgOrder(o, mkt))
+	}
```

Comment:
When I saw the EpochOrders and BookOrders declarations, my first question was "why do the order types need to be separate returns?" and it looks like they don't need to be.  Thoughts about a `OrderToMsgOrder` function that deals with the type, and returning a single `[]order.Order` slice from EpochOrders and BookOrders?

Reply:
Should I change the BookOrders return type? It only has the one.

Reply:
Yeah, good call. `LimitOrder` only for `BookOrders` return.